nssp/tests/test_pull.py

```diff
@@ -33,11 +33,27 @@ def test_pull_nssp_data(self, mock_socrata):
         mock_client.get.side_effect = [test_data, []]  # Return test data on first call, empty list on second call
         mock_socrata.return_value = mock_client
 
+        backup_dir = 'test_raw_data_backups'
+        # Clear old backup files
+        for file in os.listdir(backup_dir):
+            os.remove(os.path.join(backup_dir, file))
+
+        custom_run = False
+
+        mock_logger = MagicMock()
+
         # Call function with test token
         test_token = "test_token"
-        result = pull_nssp_data(test_token)
+        result = pull_nssp_data(test_token, backup_dir, custom_run, mock_logger)
         print(result)
 
+        # Check logger used:
+        mock_logger.info.assert_called()
```

Comment:
you can also assert on the actual log message by using caplog: (https://pytest-with-eric.com/fixtures/built-in/pytest-caplog/)
```
 def test_merge_backfill_file(self, caplog, monkeypatch):
        caplog.set_level(logging.INFO)
        logger = get_structured_logger()

        fn = "quidel_covidtest_202008.parquet"
        assert fn not in os.listdir(backfill_dir)
        
        # Check when no daily file stored
        today = datetime(2020, 8, 20)
        merge_backfill_file(backfill_dir, today, logger, test_mode=True)
        assert fn not in os.listdir(backfill_dir)
        assert "No new files to merge; skipping merging" in caplog.text
       
```